### PR TITLE
[#307] Describe releasing process and support policy

### LINKED ISSUE #307: Document releases support policy and expectations
# Clarification and motivation

We don't have a clearly defined support policy documented in the repository.

For example, it's not clear by what criteria we decide what OS versions we release native packages for in Ubuntu (last LTS? all LTS still supported by Canonical? only the last two?) or macOS.

We should document this so that users know what to expect now and, most importantly, in the future.

Additionally, especially given that we are moving to a mostly automated release process, it would be great to have a graph documenting what triggers/waits on what and how soon one might expect to see the results available. 

# Acceptance criteria

The repo contains a document explaining:
- the support policy for upcoming releases
- inter-dependencies and expected waiting times for new releases

### PR DESCRIPTION
**Pull request by rvem**
## Description
This PR adds two new docs to the repository: the first one describes the OSes versions support policy, the second describes
the releasing process in `tezos-packaging`.

## Related issue(s)

Resolves #307

#### Related changes (conditional)

- [x] I checked whether I should update the [README](/serokell/tezos-packaging/tree/master/README.md)

- [x] I checked whether native packaging works, i.e. native binary packages
  can be successfully built.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).

### CODE REVIEW

File: docs/support-policy.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+There are packages for `x86_64` and `aarch64` architectures.
+
+## macOS brew formulae
+
+Brew formulae provided by `tezos-packaging` support all maintained macOS versions.
+
+Note, that currently Octez doesn't support M1 and one have to use `rosetta` to use
+brew formulae on M1.
+
+Unfortunately, an ability to provide pre-compiled brew bottles for formulae hardly
```
Comment (itkach) [inline docs/support-policy.md:38]: hardly depends -> has a hard dependency

File: docs/support-policy.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+* 20.04 LTS (Focal Fossa)
+
+Also, we provide support for non-LTS versions on request from the users.
+E.g. 21.04 (Hirsute Hippo) support was requested in [#212](https://github.com/serokell/tezos-packaging/issues/212).
+
+There are packages for `arm64` and `amd64` architectures.
+
+## Fedora packages
+
+We provide support for all maintained Fedora versions: Fedora 34, Fedora 35.
```
Comment (ixahedron) [inline docs/support-policy.md:27]: For consistency, it's probably better to make a list out of Fedora versions too

File: README.md [hunk 3]
```diff
@@ -40,6 +40,8 @@ the binaries, and the services used.
 - [**Fedora**](./docs/distros/fedora.md)
 - [**macOS**](./docs/distros/macos.md)
 
+The information about supported versions of the aformentioned OSes is available in the [support policy doc](./docs/support-policy.md).
```
Comment (ixahedron) [inline README.md:43]:
```suggestion
The information about supported versions of the aforementioned OSes is available in the [support policy doc](./docs/support-policy.md).
```

File: docs/release-workflow.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+   - SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+   -
+   - SPDX-License-Identifier: LicenseRef-MIT-TQ
+   -->
+# Release workflow
+
+This document explains the steps and timings of the release process in `tezos-packaging`.
+
+Releasing process is almost fully automated, the only steps that require manual
+interaction are the review of two PRs created by the CI.
```
Comment (ixahedron) [inline docs/release-workflow.md:11]: I don't think it necessarily makes sense to fully transcribe what the diagram says, but I'd like at least some more info to be in text, e.g. for people using screenreaders (I'm not sure if the .svg text is supported there), or just for information at a glance. Let's maybe add just another line here about how long it usually takes, e.g. 

```suggestion
Releasing process is almost fully automated, with the initial release PR being created
at most 4 hours after a new Octez release. The only steps that require manual intervention
are reviews of this initial PR and of the PR that updates macOS brew formulae.
This allows `tezos-packaging` to closely follow Octez releases without large waiting times.
```

What do you think? 🤔

Comment (rvem) [reply docs/release-workflow.md:11]: That makes sense to mention in the doc as well

File: docs/support-policy.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+   - SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+   -
+   - SPDX-License-Identifier: LicenseRef-MIT-TQ
+   -->
+# Packages support policy
+
+`tezos-packaging` provides packages for Ubuntu, Fedora and macOS for each stable and release-candidate
+version of the Octez.
+
+This document targets to explain which versions of the aformentioned OSes are targeted
```
[the remark this answers is not in the document]
Comment (rvem) [reply docs/support-policy.md:11]: :facepalm:

File: docs/support-policy.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+Also, we provide support for non-LTS versions on request from the users.
+E.g. 21.04 (Hirsute Hippo) support was requested in [#212](https://github.com/serokell/tezos-packaging/issues/212).
+
+There are packages for `arm64` and `amd64` architectures.
+
+## Fedora packages
+
+We provide support for all maintained Fedora versions:
+* Fedora 34
+* Fedora 35
```
Comment (ixahedron) [inline docs/support-policy.md:29]: LGTM now... except that it looks like, for the time being, we're unable to support Fedora 35 after all :kekw:

**Review by ixahedron — APPROVED** [review]

File: docs/support-policy.md (hunk 6 above)
Comment (rvem) [reply docs/support-policy.md:29]: Uhmm, yes :see_no_evil:

File: docs/support-policy.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+E.g. 21.04 (Hirsute Hippo) support was requested in [#212](https://github.com/serokell/tezos-packaging/issues/212).
+
+There are packages for `arm64` and `amd64` architectures.
+
+## Fedora packages
+
+<!-- TODO: mention that we support all maintained versions once TODO below is resolved -->
+We provide support for the following Fedora versions:
+* Fedora 34
+<!-- * Fedora 35 TODO: uncomment once we're done with Fedora 35 support -->
```
Comment (pasqu4le) [inline docs/support-policy.md:30]: Continuing the same, IMO this should mention that we try to cover all [Fedora releases](https://docs.fedoraproject.org/en-US/releases/) currently supported.

I think that it's also better to remove the TODO and instead explain that Fedora 35 is currently unsupported due to [tezos/#1449](https://gitlab.com/tezos/tezos/-/issues/1449)

File: docs/support-policy.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+There are packages for `x86_64` and `aarch64` architectures.
+
+## macOS brew formulae
+
+Brew formulae provided by `tezos-packaging` support all maintained macOS versions.
```
Comment (pasqu4le) [inline docs/support-policy.md:36]: Here this sounds good to me as the "aim", but I'd also make a list of the current versions

File: docs/support-policy.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+Unfortunately, an ability to provide pre-compiled brew bottles for formulae has
+a hard dependency on the available build infrastructure. At the moment, the only macOS
+versions for which we provide brew bottles are Big Sur (both `x86_64` and `arm64`) and Catalina.
```
Comment (pasqu4le) [inline docs/support-policy.md:40]: How about putting these in a bulleted list too for consistency?

File: docs/support-policy.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+There are packages for `x86_64` and `aarch64` architectures.
+
+## macOS brew formulae
+
+Brew formulae provided by `tezos-packaging` aim to support all maintained macOS versions:
+* macOS 10.15 (Catalina)
+* macOS 11 (Big Sur)
+* macOS 12 (Monterey)
+
+Unfortunately, an ability to provide pre-compiled brew bottles for formulae has
```
Comment (pasqu4le) [inline docs/support-policy.md:46]:
```suggestion
Unfortunately, the ability to provide pre-compiled brew bottles for formulae has
```

File: docs/release-graph.xml [hunk 11]
```diff
@@ -0,0 +1 @@
...eotFF4z8g5dYEk9SSoymTOTOjsKHyk3r8KjS9zzsjy42ZWXe2pmPv1gBQ8Vok8MIWW9YQsQL5gp8hqtr/zgsMlrfAc5Biiw4CGJH0ccgPYmi26vz6SGDDBOMbAmPmfSSsNm96i90C1soIoqK8wBa/x18fEglPar6qWR2QCqNjxZUxzBkVv3VGJdyVRKO2xrQdRmcv0vhaCZsXsTGjfmA43ya96a77DHLbtMh2smfmHAnN6PJp7h1I8/CsaO5ZNF/8qNRcAJGww2ds/fbxrDjd6frJSB3swS6D5KHSZ51oJSNiuJjrGPvRSrVayejRBURiazsGajNIvcrCHmGTQ5AJo6sC2wmCDAINClyKR+lbM5DTNG1SCyr6RGI9lWJ9yWkhNTzh9SRcqrkwm6omsdTUlRT8ARac4Zb8ZcELeJ2gBm4wCKobHRjU8FhBnVtAv5oOud9BhcIDVWh2VirkesdD3Tkj1F33rGAPvyb+t1RmdYyNyfS6FPBmMl32irXGQR0zXGmCjZgWRFCors7qmPBPXvtMLZTVSWpQThVwavkK+efiL+DfmgodiZwUNWFjLo8U1gh5RHIFZhFX6s+HImmPbF2WDkftaXJA5uJaFgqArs6VZGW76iq3rKtspMa9kDNpPmSQc2oGzcbztKxjRjEOKiB/xHUha/XyQlHqHaRcEI1E8oBXZOWCAuk570ldJFlJ0s51wUtx0owNomHGhv4I3uH3xLv99rGbsj2OuNcd8FM7Q7p8dsi9Ir1JJGeQOc8f0tGRGWjxL5Hnkovtle33e6anqSuUA7bVcz80y1KawusSQ2g9pK+ppoxseBDXlGkOaDSqOs6plGPr6gl0kbk+G3IvGLtUTEe4Nz8a90I7GXeuymbr/7O7sXvoN6D2OnAmBZJrfwXSea2zp+okFJOOE90fEQOhj9KYS8l6GW4SvSuvTFF11eoGL6Go2vNbZkTql6SmKjNT4vUyrxkB+3Q/oZi78wPEfCyhjifme67pPZ45FUJf1XvhPSmGXqsBBsNoduoCxH3VGtZpitbLPFG86fCLRuSFdvCCkeD5Rwve3AredadGTTGj9Ey2cvY8go26DaPcVjZWFWMXOlrwLL//Xuh87SpD9NmI/tGemsdS6A6M5oNdUxL16nqJLO0k+KW6Z4ylrvPtNMVu/182Pbbzv0r/5gs=</diagram></mxfile>
```
[the remark this answers is not in the document]
Comment (pasqu4le) [reply docs/release-graph.xml]: @rvem A few comments on the current version:
1. There are two "create release PR" blocks, I suppose that the second/lower one should be replaced with a block about building binaries. Or perhaps removed and more binary-building blocks added below?
2. There are steps like "Create new release with static binaries" that are about the GitHub releases, but this isn't clarified so it's confusing what "release" here is meaning. I think we should add "GitHub" in these.
Perhaps it's worth also adding "Octez" in all the blocks were the word release is used to mean that one too. 
3. I see that there is a single block about building brew bottles, I wonder if isn't worth giving more details there, about the fact that it's currently divided in steps and/or executed on two different infrastructures. What do you think?
4. The two blocks on the bottom say the same thing, but one should not be about the stable mirror.

[the remark this answers is not in the document]
Comment (rvem) [reply docs/release-graph.xml]: > There are two "create release PR" blocks, I suppose that the second/lower one should be replaced with a block about building binaries. Or perhaps removed and more binary-building blocks added below?

Actually, seems like the second "Create release PR" is redundant

[the remark this answers is not in the document]
Comment (rvem) [reply docs/release-graph.xml]: > I see that there is a single block about building brew bottles, I wonder if isn't worth giving more details there, about the fact that it's currently divided in steps and/or executed on two different infrastructures. What do you think?

Well, this is highly likely going to change over time and it'll be easy to forget to update it here

[the remark this answers is not in the document]
Comment (pasqu4le) [reply docs/release-graph.xml]: OK, fair enough. I suppose it's not very useful to get into such levels of details :+1:

**Review by pasqu4le — APPROVED** [review]

**Pull request merged**